### PR TITLE
add specs

### PR DESCRIPTION
**Pull request by jsvd**
addresses #2

### CODE REVIEW

File: spec/inputs/graphite_spec.rb [hunk 1]
```diff
@@ -1 +1,54 @@
...
+    srand(RSpec.configuration.seed)
+  end
+
+  let(:host) { "127.0.0.1" }
+  let(:port) { rand(5000) + 1025 }
+  let(:queue) { [] }
+
+  let(:client) { TCPSocket.new(host, port) }
+
+  subject { LogStash::Inputs::Graphite.new("host" => host, "port" => port) }
```
Comment (purbon) [inline spec/inputs/graphite_spec.rb]: I wonder why not using the kind of input helper @colinsurprenant added in devutils? https://github.com/elastic/logstash-devutils/blob/master/lib/logstash/devutils/rspec/logstash_helpers.rb#L64

Comment (jsvd) [inline spec/inputs/graphite_spec.rb]: The biggest reason is to not run code that is not related to this plugin. Using that will make this test depend on the pipeline

Comment (purbon) [inline spec/inputs/graphite_spec.rb]: I understand, but we should otherwise try to standardize the way we start the pipeline, can you wrap this thing you did here in a helper alike to the other? This would be super helpful to others.

Comment (jsvd) [inline spec/inputs/graphite_spec.rb]: this is very specific to his plugin: initialize it with parameters, and use a tcpsocket to write the same address

Comment (purbon) [inline spec/inputs/graphite_spec.rb]: are you sure you can not make it somehow generic? there are others like https://github.com/logstash-plugins/logstash-input-ganglia that would be on the same page, isn't?

I would like to avoid us implementing the same pattern over and over.

Comment (jsvd) [inline spec/inputs/graphite_spec.rb]: I don't see how to dry up this more. if using the whole pipeline is the way to dry a couple of lines of set up code, I'm against it.

**Comment by talevy** [comment]
should this be merged?

**Comment by suyograo** [comment]
@purbon ping. can we reach a decision here and make progress?

**Comment by purbon** [comment]
@suyograo as I told @jsvd it would better to wrap the was this test handles plugin management in a helper method as is in other plugins, this is not necessary to be done by invoking the pipeline, but would make all this code easier to understand and manage.

**Comment by colinsurprenant** [comment]
if these tests are valid tests, I say we merge them. valid tests will be better than no test. we can open a new issue with test refactor ideas. I would really love to have tests on this for the shutdown refactor and 2.0 release. any objection @purbon ?

**Comment by suyograo** [comment]
its really sad to see a test PR stalled.. +1 on merging this in to have *some* tests

**Comment by colinsurprenant** [comment]
we can close this, it will be replaced by #5 - the commit here has been moved in #5
